csharp/msbuild/CodeAnalysis.Base.globalconfig

```diff
@@ -264,6 +291,102 @@ dotnet_diagnostic.CA1859.severity = none
 
 # TODO review rules below
 
```

Comment:
Most of the rules below must be enabled, and fix the warnings but we can do in separate PRs.